### PR TITLE
Add necessary comments to the code #39

### PR DESCRIPTION
**Pull request by chavi362**

### CODE REVIEW

File: KeyFlow/helper.hpp [hunk 1]
```diff
@@ -10,13 +10,13 @@ class Helper
...
-
+	//constrctor
 	Helper()
 	{
-		init_key_to_vkcode();
+		init_key_to_vkcode();//intialized the unordered_map
 		Logger::log("Helper()\n");
 	}
-
+	//removes the removes leading and trailing spaces from str
```
Comment (hetulbhatt) [inline KeyFlow/helper.hpp]: remove the extra "remove"

File: KeyFlow/helper.hpp [hunk 2]
```diff
@@ -70,7 +71,9 @@ class Helper
 		}
 		return fullString.substr(0, prefix.length()) == prefix;
 	}
-
+	//nitializes the key_to_vkcode unordered map using an initializer list.
```
Comment (hetulbhatt) [inline KeyFlow/helper.hpp]: Correct typo: "nitializes"

File: KeyFlow/helper.hpp [hunk 3]
```diff
@@ -70,7 +71,9 @@ class Helper
 		}
 		return fullString.substr(0, prefix.length()) == prefix;
 	}
-
+	//nitializes the key_to_vkcode unordered map using an initializer list. 
+	//The initializer list contains a series of key-value pairs enclosed in curly braces {}.
+	//Each key-value pair corresponds to a key (represented as a std::string) and a value (represented as an int) according to the table in the Readmy
```
Comment (hetulbhatt) [inline KeyFlow/helper.hpp]: Correct typo: "Readmy"

File: KeyFlow/keyhook.hpp [hunk 4]
```diff
@@ -57,18 +60,19 @@ class KeyHook
...
 	KeyHook* hook = nullptr;
-
+	//The constructor initializes the hook member variable by creating a new KeyHook object
 	KeyHookSingleton()
 	{
 		hook = new KeyHook();
 	}
 
 public:
+	//provides the only way to access the single instance of the KeyHook class. It returns a reference to the hook member variable of the KeyHookSingleton instanc
```
Comment (hetulbhatt) [inline KeyFlow/keyhook.hpp]: Correct typo: "instanc"

File: KeyFlow/keystroke_handler.hpp [hunk 5]
```diff
@@ -22,6 +22,8 @@ class KeystrokeHandler
 	std::unordered_map<std::string, int> key_to_vkcode;
 
 public:
+	//nstructor that initializes the sequence variable to 0 and sets all elements of the logs array to 0 using memset.
```
Comment (hetulbhatt) [inline KeyFlow/keystroke_handler.hpp]: Correct typo: "nstructor"

File: KeyFlow/keystroke_handler.hpp [hunk 6]
```diff
@@ -33,7 +35,7 @@ class KeystrokeHandler
 	{
 		return this->loader;
 	}
-
+	//akes an integer size as a parameter and resizes the buffer vector accordingly.
```
Comment (hetulbhatt) [inline KeyFlow/keystroke_handler.hpp]: Correct typo: "akes"

File: KeyFlow/keystroke_handler.hpp [hunk 7]
```diff
@@ -114,6 +116,8 @@ class KeystrokeHandler
 			}
 		}
 	}
+	//handle keystroke events.
+	//- It takes a DWORD parameter vk_code representing the virtual key code of the pressed key and an int parameter isDown indicating whether the key is pressed or released.
```
Comment (hetulbhatt) [inline KeyFlow/keystroke_handler.hpp]: Remove the hyphen before " It takes..."

File: KeyFlow/logger.hpp [hunk 8]
```diff
@@ -1,18 +1,19 @@
...
 #include <iostream>
 #include <fstream>
-#include <filesystem>
+#include <filesystem>//provides access to file and directory operations
 
 class Logger
 {
 private:
-	static std::ofstream logFile;
+	static std::ofstream logFile;//his member variable will be used to write log messages to a file
```
Comment (hetulbhatt) [inline KeyFlow/logger.hpp]: Correct typo: "his"

File: KeyFlow/logger.hpp [hunk 9]
```diff
@@ -1,18 +1,19 @@
...
 
 public:
+	//his static member function is used to initialize the logger. It creates a directory named "logs" (if it doesn't already exist) using the _mkdir() function from <direct.h>.
```
Comment (hetulbhatt) [inline KeyFlow/logger.hpp]: Correct typo: "his"

File: KeyFlow/KeyFlow.cpp [hunk 10]
```diff
@@ -1,8 +1,10 @@
 #include "keyhook.hpp"
 
 int main(int argc, char* argv[]) {
+	///initialize the logger. It creates a directory named "logs" (if it doesn't already exist) then opens the file for writing
```
Comment (hetulbhatt) [inline KeyFlow/KeyFlow.cpp]: Remove an extra "/" from the comment

**Review by hetulbhatt — CHANGES_REQUESTED** [review]
Great job, @chavi362! Your comments look good overall. However, I have a few PR-level comments for you:

1. **Consistent Capitalization in Comments**
   - Please make sure that the capitalization in your comments is consistent throughout the codebase. For example:
     - Good: `// Initialize the variable.`
     - Bad: `// initialize the variable.`

2. **Spacing After "//" in Comments**
   - When writing comments, make sure to leave a space after the "//" characters. For instance:
     - Good: `// This is a comment.`
     - Bad: `//This is a comment.`

3. **Spacing Before Comments on the Same Line as Code**
   - If you place a comment on the same line as the code, make sure to leave a space before starting the comment. For example:
     - Good:
       ```cpp
       int x = 5;  // Initialize x.
       ```
     - Bad:
       ```cpp
       int x = 5;//Initialize x.
       ```

4. **End Comments with a Full Stop**
   - Please remember to end your comments with a full stop. It helps in making comments more complete and clear. For instance:
     - Good: `// This is a comment.`
     - Bad: `// This is a comment`

Please address these points in your code. Thanks!

**Comment by hetulbhatt** [comment]
@chavi362, just let me know once you resolve all the issues, and don't forget point 3. Thanks.

"3. Spacing Before Comments on the Same Line as Code"

**Comment by chavi362** [comment]
Fixed! Yes I did it! thank you for the review, I've learned a lot

‫בתאריך יום א׳, 10 בספט׳ 2023 ב-20:40 מאת ‪Hetul Bhatt‬‏ <‪
***@***.***‬‏>:‬

**Comment by hetulbhatt** [comment]
@chavi362, merging this PR then. Thanks for the efforts. Please close the issue assigned to you. You can start looking into some other issue too, if you want to, as you're now well acquainted with the code base.

**Comment by chavi362** [comment]
Thanks!!
I would like to work on another issue,
but now I see that all the issues currently assigned to others.
Is there any issue I could work on?

‫בתאריך יום א׳, 10 בספט׳ 2023 ב-20:51 מאת ‪Hetul Bhatt‬‏ <‪
***@***.***‬‏>:‬

**Comment by hetulbhatt** [comment]
Just created a new issue. It's a bug this time. Please have a look: https://github.com/hetulbhatt/KeyFlow/issues/45

**Pull request merged**